### PR TITLE
CSI E2E backupRestore test for cluster-profile Azure on 4.10+

### PR DESCRIPTION
**Pull request by hhpatel14**
Default CSI driver is not available in Azure OCP versions < 4.10. Therefore, CSI BackRestore tests are skipped for versions < 4.10

### CODE REVIEW

File: tests/e2e/backup_restore_suite_test.go [hunk 1]
```diff
@@ -104,6 +105,14 @@ var _ = Describe("AWS backup restore tests", func() {
 				Skip(reason)
 			}
 
+			if provider == "azure" && brCase.BackupRestoreType == CSI {
+				availability, _ := version.NewVersion("4.10")
```
Comment (weshayutin) [inline tests/e2e/backup_restore_suite_test.go:109]: can we make this >=("4.10") before we move on?

**Review by weshayutin — CHANGES_REQUESTED** [review]
let's be sure we test 4.10 and greater for azure csi :)

**Comment by weshayutin** [comment]
2022/07/27 15:44:30 Creating backup mysql-csi-e2e-e99110a8-0dc2-11ed-99b8-0a580a834e77 for case mysql-csi-e2e
2022/07/27 15:45:12 Backup for case mysql-csi-e2e succeeded
2022/07/27 15:45:12 waiting for volume snapshot contents snapcontent-02d6b848-cd4d-48f8-8d0d-e1eb6fe1aafd to be ready
2022/07/27 15:45:12 Uninstalling app for case mysql-csi-e2e
2022/07/27 15:45:22 Creating restore mysql-csi-e2e-e99110ac-0dc2-11ed-99b8-0a580a834e77 for case mysql-csi-e2e
checking for the NAMESPACE: mysql-persistent
 2022/07/27 15:45:54 Running post-restore function for case mysql-csi-e2e
2022/07/27 15:45:54 Verifying backup/restore data of todolist
I0727 15:45:55.679320   11905 request.go:665] Waited for 1.049470293s due to client-side throttling, not priority and fairness, request: GET:https://api.ci-op-wpn7gjl2-d24fb.ci.azure.devcluster.openshift.com:6443/apis/apiextensions.k8s.io/v1?timeout=32s
Data came from backup-file
 [{"Id":1,"Description":"time to make the donuts","Completed":false},{"Id":3,"Description":"2022-07-27 15:44:30.341667912 +0000 UTC m=+674.447116404","Completed":false},{"Id":4,"Description":"Wednesday","Completed":false}]

Data from the response after restore
 [{"Id":1,"Description":"time to make the donuts","Completed":false},{"Id":3,"Description":"2022-07-27 15:44:30.341667912 +0000 UTC m=+674.447116404","Completed":false},{"Id":4,"Description":"Wednesday","Completed":false}]

IT WORKED :)
Thank you!  One nit on version checker need >= 4.10

File: tests/e2e/backup_restore_suite_test.go [hunk 2]
```diff
@@ -104,6 +104,15 @@ var _ = Describe("AWS backup restore tests", func() {
 				Skip(reason)
 			}
 
+			if provider == "azure" && brCase.BackupRestoreType == CSI {
+				if brCase.MinK8SVersion == nil {
+					brCase.MinK8SVersion = &K8sVersion{Major: "1", Minor: "24"}
```
Comment (kaovilai) [inline tests/e2e/backup_restore_suite_test.go:109]: version >= 4.10 should be min k8s version 1.23
```suggestion
					brCase.MinK8SVersion = &K8sVersion{Major: "1", Minor: "23"}
```

Comment (kaovilai) [inline tests/e2e/backup_restore_suite_test.go:109]:
```suggestion
					brCase.MinK8SVersion = &K8sVersion{Major: "1", Minor: "23"}
```
So >= 4.10 should be min version 1.23

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: tests/e2e/backup_restore_suite_test.go [hunk 3]
```diff
@@ -104,6 +104,15 @@ var _ = Describe("AWS backup restore tests", func() {
...
 			}
 
+			if provider == "azure" && brCase.BackupRestoreType == CSI {
+				if brCase.MinK8SVersion == nil {
+					brCase.MinK8SVersion = &K8sVersion{Major: "1", Minor: "23"}
+				}
+				if notVersionTarget, reason := NotServerVersionTarget(brCase.MinK8SVersion, brCase.MaxK8SVersion); notVersionTarget {
+					Skip(reason)
+				}
+			}
```
Comment (kaovilai) [inline tests/e2e/backup_restore_suite_test.go:114]:
```suggestion
			if provider == "azure" && brCase.BackupRestoreType == CSI {
				if brCase.MinK8SVersion == nil {
					brCase.MinK8SVersion = &K8sVersion{Major: "1", Minor: "23"}
				}
			}
			if notVersionTarget, reason := NotServerVersionTarget(brCase.MinK8SVersion, brCase.MaxK8SVersion); notVersionTarget {
				Skip(reason)
			}
```

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: tests/e2e/backup_restore_suite_test.go (hunk 3 above)
Comment (kaovilai) [reply tests/e2e/backup_restore_suite_test.go:114]: The server version check code should be outside `if provider == "azure"`

File: tests/e2e/backup_restore_suite_test.go (hunk 1 above)
Comment (hhpatel14) [reply tests/e2e/backup_restore_suite_test.go]: For anything >= 4.10 will not  hit the skip() statement. I do not think we need >=4.10 condition.

**Comment by kaovilai** [comment]
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_oadp-operator/771/pull-ci-openshift-oadp-operator-master-4.10-operator-e2e-azure/1552729425278144512/artifacts/operator-e2e-azure/e2e/build-log.txt shows azure CSI tests being run on 4.10
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_oadp-operator/771/pull-ci-openshift-oadp-operator-master-4.9-operator-e2e-azure/1552729436141391872/artifacts/operator-e2e-azure/e2e/build-log.txt shows 4.9 skipping CSI.

**Review by kaovilai — APPROVED** [review]

**Pull request merged**